docs/components/index.md

```diff
@@ -0,0 +1,360 @@
+
+A child component can receive property assignments from its calling parent. In the following example, the `ParentComponent` sets the value of the `Title` property in the `ChildComponent`:
+
+*ParentComponent.cshtml*:
+
+[!code-cshtml[](common/samples/2.x/ComponentsSample/Pages/ParentComponent.cshtml?start=1&end=7)]
+
+*ChildComponent.cshtml*:
+
+[!code-cshtml[](common/samples/2.x/ComponentsSample/Pages/ChildComponent.cshtml)]
+
+The body of the Bootstrap-styled panel is provided by `ChildContent`. `ChildContent` is loaded where the `<ChildComponent>` element appears in the parent component.
+
+## Routing
+
+Routing in Blazor is achieved by providing a route template to each accessible component in the app.
+
+When a *\*.cshtml* file with an `@page` directive is compiled, the generated class is given a [RouteAttribute](https://docs.microsoft.com/dotnet/api/microsoft.aspnetcore.mvc.routeattribute) specifying the route template. At runtime, the router looks for component classes with a `RouteAttribute` and renders whichever component has a route template that matches the requested URL.
+
+Multiple route templates can be applied to a component. The following component responds to requests for `/BlazorRoute` and `/DifferentBlazorRoute`:
+
+[!code-cshtml[](common/samples/2.x/ComponentsSample/Pages/BlazorRoute.cshtml?start=1&end=4)]
+
+## Route parameters
+
+Blazor components can receive and use route parameters from the route template provided in the `@page` directive:
+
+*RouteParameter.cshtml*:
+
+[!code-cshtml[](common/samples/2.x/ComponentsSample/Pages/RouteParameter.cshtml?start=1&end=8)]
+
+Not all route parameter constraints are supported at this time. Optional parameters aren't currently supported, so two `@page` directives are applied in the example above. The first permits navigation to the component without a parameter. The second `@page` directive takes the `{text}` route parameter and assigns the value to the `Text` property.
+
+## JavaScript/TypeScript interop
+
+To call JavaScript libraries or custom JavaScript/TypeScript code from .NET, the current approach is to register a named function with JavaScript/TypeScript. Place the `registerFunction` call in the *index.html* file or a JavaScript file (*\*.js*) loaded by the *index.html* file. Place the inline JavaScript or `<script>` tag below `<script type="blazor-boot"></script>`, and the JavaScript/TypeScript loads at the correct time and only executes once. 
+
+```javascript
+Blazor.registerFunction('doPrompt', function(message) {
+    return prompt(message);
+});
+```
+
+Wrap the named function for calls from .NET:
+
+```csharp
+public static bool DoPrompt(string message)
+{
+    return RegisteredFunction.Invoke<bool>("doPrompt", message);
+}
+```
+
+This approach has the benefit of working with JavaScript build tools, such as [webpack](https://webpack.js.org/).
+
+The Mono team is working on a library that exposes standard browser APIs to .NET.
```

Comment:
I think we're missing sections on defining component parameters and also how to use components (i.e. `<Counter IncrementAmount="10" />`). Also, in the using components section I think we need to document that you have to use `@addTagHelper` to use a component that lives in a different assembly than the app assembly.

Reply:
The section is there, but I called them "route parameters" (`## Route parameters`) ... shall I change the name to "component parameters"?

Reply:
There's a difference between the two:

* Component parameters are what you can pass in using HTML attribute syntax, e.g., `<MyComponent SomeValue=@Something />`
* Route parameters are the tokens in a route pattern, e.g., `@page "/products/{productId}"`.

The router uses route parameters to populate the corresponding component parameters. So they are related, but different concepts.